src/mod_websockets.erl

```diff
@@ -218,8 +235,15 @@ maybe_start_fsm([#xmlstreamstart{ name = <<"stream", _/binary>>, attrs = Attrs}
     end;
 maybe_start_fsm([#xmlel{ name = <<"open">> }],
                 #ws_state{fsm_pid = undefined} = State) ->
-    Opts = #{access => all, shaper => none, xml_socket => true, hibernate_after => 0},
-    do_start_fsm(ejabberd_c2s, Opts, State);
+    Opts = #{
+        access => all,
+        shaper => none,
+        max_stanza_size => 0,
+        xml_socket => true,
+        hibernate_after => 0,
+        c2s_state_timeout => 5000,
+        backwards_compatible_session => true},
```

Comment:
Here's a bit more important to have the correct config, websockets are ubiquitous. You can check them on the listener, using some `mongoose_config` function to get the listener config and then extract them. Or maybe better, they should be kept in the websocket state, `#ws_state{}`, from there we can correctly configure this.